[en]: update multus-underlay/dpdk (#5167)
DaoCloud/DaoCloud-docs: docs/en/docs/network/modules/multus-underlay/dpdk.md

---
MTPE: WANG0608GitHub
DATE: 2024-08-02
---

# DPDK

This page mainly introduces how to quickly create the first DPDK application in DCE 5.0.

> ## Pre-dependency

Reviewer: ```suggestion
## Prerequisites and dependencies
```

> - Install Multus-underlay, and enable the installation of SRIOV components, refer to [Install](install.md)

Reviewer: ```suggestion
- Multus-underlay has been installed and SRIOV is enabled, refer to [Installation](install.md)
```

> - Hardware support is required: have a NIC that supports SR-IOV series and set up a virtual feature (VF), refer to [SR-IOV](sriov.md)

Reviewer: ```suggestion
- Hardware: a NIC that supports SR-IOV, with virtual function (VF) enabled, refer to [SR-IOV](sriov.md)
```

> - Need to switch the NIC driver to user mode driver

Reviewer: ```suggestion
- The NIC driver has been adapted to the user mode
```

Take `0000:04:00.2 'MT27800 Family [ConnectX-5 Virtual Function] 1018' if=enp4s0f0v0 drv=mlx5_core unused=vfio-pci` as an example:

- 0000:04:00.2: The VF PCI address
- if=enp4s0f0v0: The VF NIC name
- drv=mlx5_core: The current NIC driver

> - unused=vfio-pci: The switchable NIC driver

Reviewer: ```suggestion
- unused=vfio-pci: An NIC driver that is ready to use if necessary
```

There are three types of user mode drivers supported by DPDK:

- vfio-pci: When IOMMU is enabled, this driver is recommended for best performance and security

> - igb-uio: It is More applicable than uio_pci_generic, and supports SR-IOV VF, but needs to manually compile the module and load it into the kernel

Reviewer: ```suggestion
- igb-uio: It is more applicable than uio_pci_generic, and supports SR-IOV VF. However, it is rquired to manually compile modules and load them into the kernel
```

    !!! note

        To update the above configuration, you need to restart the system, it is best to back up before restarting the system.

>         If the configuration cannot be updated, the driver needs to be switched to the igb-uio driver,
>         and manual build && insmod && modprobe is required. For details, refer to [dpdk-kmod](https://github.com/atsgen/dpdk-kmod)

Reviewer: ```suggestion
        If the configuration cannot be updated, the driver needs to be switched to the igb-uio driver.
        You need to manually perform actions like build, insmod, and modprobe.
        For details, refer to [dpdk-kmod](https://github.com/atsgen/dpdk-kmod).
```

> EOF
```

> Wait for pod Running, then enter the pod:

Reviewer: ```suggestion
Wait for the pod to be running, then enter it by running the following command:
```

    IfName="net1" Name="kube-system/sriov-dpdk-vlan0" Type=SR-IOV
    MAC=""

myArgc=14
dpdk-app -n 4 -l 1 --master-lcore 1 -w 0000:04:01.1 -- -p 0x1 -P --config="(0,0,1)" --parse-ptype
```

> DPDK-app will print out the relevant information of the current pod, including the IP, MAC and type of eth0.

Reviewer: ```suggestion
dpdk-app will print out the relevant information of the current pod, including the IP, MAC and type of eth0.
```
dpdk-app 是上面代码中提到的，不能乱改
